### PR TITLE
Fix immutable baggage when returning iterator from context

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java [hunk 1]
```diff
@@ -64,7 +63,7 @@ public SpanContext(long traceID, long spanID, long parentID, byte flags) {
 
   @Override
   public Iterable<Map.Entry<String, String>> baggageItems() {
-    return baggage.entrySet();
+    return new HashMap<String, String>(baggage).entrySet();
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java]: not quite sure where you're going with this. The main immutability of the context (and baggage by extension) comes from this line in the [setBaggageItem](https://github.com/uber/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/Span.java#L135), i.e. the whole context is replaced when baggage is amended.

```java
this.context = this.context.withBaggageItem(key, value);
```

Comment (pavolloffay) [inline jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java]: Iterator in Java allows to remove an item from the underlying collection.

Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java]: ok, I am not too concerned with this method, it's rarely used anyway. The potential future improvement would be to use immutable map in the constructor.

**Comment by yurishkuro** [comment]
can't merge due to a conflict

**Comment by pavolloffay** [comment]
rebased

**Pull request merged**